app/lib/common/widgets/scrollable_list_tab_scroller.dart

```diff
@@ -0,0 +1,428 @@
+          ),
+        ),
+        buildCustomBodyContainerOrDefault(
+          context: context,
+          child: Builder(
+            builder: (context) {
+              WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+                final size = context.size;
+                if (size != null) {
+                  _currentPositionedListSize = size;
+                }
+              });
+              return ScrollsToTop(
+                onScrollsToTop: _onScrollsToTop,
+                child: ScrollablePositionedList.builder(
+                  itemBuilder: (a, b) {
+                    return widget.itemBuilder(a, b);
+                  },
+                  itemCount: widget.itemCount,
+                  itemScrollController: itemScrollController,
+                  itemPositionsListener: itemPositionsListener,
+                  shrinkWrap: widget.shrinkWrap,
+                  initialScrollIndex: widget.initialScrollIndex,
+                  initialAlignment: widget.initialAlignment,
+                  scrollDirection: widget.scrollDirection,
+                  reverse: widget.reverse,
+                  physics: widget.physics,
+                  semanticChildCount: widget.semanticChildCount,
+                  padding: widget.padding,
+                  addSemanticIndexes: widget.addSemanticIndexes,
+                  addAutomaticKeepAlives: widget.addAutomaticKeepAlives,
+                  addRepaintBoundaries: widget.addRepaintBoundaries,
+                  minCacheExtent: widget.minCacheExtent,
+                  scrollOffsetController: widget.scrollOffsetController,
+                  scrollOffsetListener: widget.scrollOffsetListener,
+                ),
+              );
+            },
+          ),
+        ),
+      ],
+    );
+  }
+
+  @override
+  void dispose() {
+    _debounce?.cancel();
+    itemPositionsListener.itemPositions.removeListener(_itemPositionListener);
+    super.dispose();
+  }
+}
+
+class DefaultHeaderWidget extends StatefulWidget {
+  final ValueNotifier<int> selectedTabIndex;
+  final IndexedActiveStatusWidgetBuilder tabBuilder;
+  final IndexedVoidCallback onTapTab;
+  final int itemCount;
+  final TabAlignment? tabAlignment;
+
+  const DefaultHeaderWidget({
+    super.key,
+    required this.selectedTabIndex,
+    required this.tabBuilder,
+    required this.onTapTab,
+    required this.itemCount,
+    this.tabAlignment,
+  });
+
+  @override
+  State<DefaultHeaderWidget> createState() => _DefaultHeaderWidgetState();
+}
+
+class _DefaultHeaderWidgetState extends State<DefaultHeaderWidget>
+    with SingleTickerProviderStateMixin {
+  late final TabController _tabController;
+
+  @override
+  void initState() {
+    super.initState();
+    _tabController = TabController(
+      length: widget.itemCount,
+      vsync: this,
+    );
+    _tabController.addListener(tabChangeListener);
+    widget.selectedTabIndex.addListener(externalTabChangeListener);
+  }
```

Comment:
I understand the scrollbable part but not why we even use a `TabController` ... We don't have tabs and there is no semantic benefit of using the tab controller. In fact, it is that very tab-controller constraint that makes the other provider so ugly (having to generate `TabEntry`-objects). Why not just have a row of custom widgets with an onTap and a background that slides around? (This is how the bottom navigation is implemented ... it isn't a tabController internally...)

Reply:
This entire file is part of third-part library code.
Library had 1 or 2 glitches which I fixed it by bringing entire file here.